app/scripts/services/import-service.js

```diff
@@ -183,8 +184,36 @@
             if (option === importServiceConflictModal.REPLACE_FILE) {
               var path = ramlRepository.join(directory.path, name);
               var file = ramlRepository.getByPath(path);
+              var promise;
 
-              file.doc.setValue(contents);
+              if (file) {
```

Comment:
1. Redundant check as at this point file already exists
2. If `file` is falsy, it means `promise` is going to stay undefined and so `promise.then` will fail, but when you take care of p.1, it will resolve by itself.